packages/payment-experience-api/src/api/paytrailMitChargeNotifyController.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AbstractController,
+  ExperienceError,
+  logger,
+  StatusCode,
+  ValidatedRequest,
+} from '@verkkokauppa/core'
+import * as yup from 'yup'
+import type { Response } from 'express'
+import {
+  parseMerchantIdFromFirstOrderItem,
+  validateApiKey,
+} from '@verkkokauppa/configuration-backend'
+import {
+  createAccountingEntryForOrder,
+  getOrderAdmin,
+} from '@verkkokauppa/order-backend'
+import { getProductAccountingBatch } from '@verkkokauppa/product-backend'
+import { sendReceipt } from '../lib/sendEmail'
+
+const requestSchema = yup.object().shape({
+  headers: yup.object().shape({
+    'api-key': yup.string().required(),
+    namespace: yup.string().required(),
+  }),
+  query: yup.object().shape({
+    orderId: yup.string().required(),
+  }),
+})
+
+export class PaytrailMitChargeNotifyController extends AbstractController<
+  typeof requestSchema
+> {
+  protected readonly requestSchema = requestSchema
+
+  protected async implementation(
+    req: ValidatedRequest<typeof requestSchema>,
+    res: Response
+  ) {
+    const {
+      headers: { 'api-key': apiKey, namespace },
+      query: { orderId },
+    } = req
+
+    await validateApiKey({ namespace, apiKey })
+
+    const order = await getOrderAdmin({ orderId })
+    const merchantId = parseMerchantIdFromFirstOrderItem(order)
+
+    if (!merchantId) {
+      logger.error('Paytrail: No merchantId found from order')
+      throw new ExperienceError({
+        code: 'merchant-id-not-found',
+        message: 'No merchantId found from order.',
+        responseStatus: StatusCode.NotFound,
+        logLevel: 'info',
+      })
+    }
+
+    const productAccountings = await getProductAccountingBatch({
+      productIds: order.items.map((item) => item.productId),
+    })
+
+    await createAccountingEntryForOrder({
+      orderId,
+      dtos: order.items.map((item) => {
+        const productAccounting = productAccountings.find(
+          (accountingData) => accountingData.productId === item.productId
+        )
+        if (!productAccounting) {
+          throw new ExperienceError({
```

Comment:
Pitäiskö tähän ja tuohon ylempään ExperienceErroriin laittaa ensin meille error notifikaatin lähetys? Nämähän taitaa olla tilanteita joissa PaymentPaid on lähetetty asiointipalvelulle mutta tiliöinti jäis tekemättä?

Reply:
Core lähettää virhetilanteissa error notifikaation. Halutaanko tupla notifikaatiot?

Reply:
Lähettääkö jos kaikki muu toimii mutta product accountingit puuttuu?